src/jit/emitxarch.cpp

```diff
@@ -12530,6 +12620,15 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
             sz = emitSizeOfInsDsc(id);
             break;
 
+        case IF_MWR_RRD_CNS:
```

Comment:
I think the register encoding needs to happen in the case for `MWR`. `AWR` is one of the only ones that handles the register encoding in itself.

Reply:
Sorry, I do not quite understand.

Reply:
If you look at the other examples MWR cases handle the encodeReg3456 before the emit Output call. emitOutputAM is the only one that does that in the call itself

Reply:
I looked the above section of `IF_RWR_MRD_CNS` that does not handle the encodeReg3456 in `if (Is4ByteSSE4OrAVXInstruction(ins))` path.

Reply:
Ah, I see.

A comment (or explicit assert) that `Is4ByteSSE4OrAVXInstruction(ins)` is always true would be beneficial.

Reply:
Thanks, will do.

Reply:
Done. Added a comment blow.